sql/analyzer/resolve_columns.go

```diff
@@ -38,6 +38,31 @@ func checkAliases(ctx *sql.Context, a *Analyzer, n sql.Node) (sql.Node, error) {
 	return n, err
 }
 
+func checkDistinctNoTuples(ctx *sql.Context, a *Analyzer, node sql.Node) (sql.Node, error) {
```

Comment:
I don't know if it makes sense here, but did you test with wrapped nodes or subqueries?
Maybe we need to inspect distinct nodes recursively here, e.g. 
https://github.com/src-d/go-mysql-server/blob/master/sql/analyzer/assign_indexes.go#L43